Migration notes for StringExpression (#6846)
cakephp/docs: en/appendices/4-2-migration-guide.rst

      for more information.
- ``Query::isResultsCastingEnabled()`` was added to get the current result
  casting mode.

> - ``StringExpression`` was added with support for setting the collation for string. 

Reviewer: ```suggestion
- ``StringExpression`` was added with support to set the collation of string literals. 
```
